include/dd/Node.hpp

```diff
@@ -28,10 +29,11 @@ namespace dd {
     inline vNode vNode::terminalNode{{{{nullptr, Complex::zero}, {nullptr, Complex::zero}}}, nullptr, 0U, -1};
 
     struct mNode {
-        std::array<Edge<mNode>, NEDGE> e{};           // edges out of this node
-        mNode*                         next{};        // used to link nodes in unique table
-        RefCount                       ref{};         // reference count
-        Qubit                          v{};           // variable index (nonterminal) value (-1 for terminal)
+        std::array<Edge<mNode>, NEDGE> e{};    // edges out of this node
+        mNode*                         next{}; // used to link nodes in unique table
+        RefCount                       ref{};  // reference count
+        Qubit                          v{};    // variable index (nonterminal) value (-1 for terminal)
+        std::int_fast8_t               flags = 0;
```

Comment:
I changed the type of the variable flags to std::uint8_t.

The advantage I see in handling booleans as integers is the efficiency and convenience since I also use node pointers to store information in a similar format. Therefore I would like to keep the format

I see your point about mixing the flags. However, before I change the boolean flags to integer flags I would like to talk to you in person.  As of right now, I would rather keep it as it is, in order to not introduce any bugs or break something. I added a corresponding todo in the code.

Reply:
As noted above, `std::uint8_t` is not guaranteed to be available in general. You would have to use `std::uint_least8_t`.
Keeping the flags is fine with me, as long as convenience functions as the one shown above are introduced.

replacing `sym` and `ident` with some bits in the `flags` register should be a very small change, given that the above mentioned convenience functions are introduced. It would just imply that any call to `p->sym` is replaced by `p->sym()` and similarly for `ident`.

Reply:
> As noted above, `std::uint8_t` is not guaranteed to be available in general. You would have to use `std::uint_least8_t`.
mb, I changed it to `std::uint_least8_t`

> replacing sym and ident with some bits in the flags register should be a very small change, given that the above mentioned convenience functions are introduced. It would just imply that any call to p->sym is replaced by p->sym() and similarly for ident.

Done. I even added set-methods as well